pkg/server/plugin/bundlepublisher/awss3/client.go

```diff
@@ -25,9 +26,20 @@ func newAWSConfig(ctx context.Context, c *Config) (aws.Config, error) {
 		cfg.Credentials = credentials.NewStaticCredentialsProvider(c.AccessKeyID, c.SecretAccessKey, "")
 	}
 
+	if c.Endpoint != "" {
+		cfg.BaseEndpoint = aws.String(c.Endpoint)
+	}
+
 	return cfg, nil
 }
 
 func newS3Client(c aws.Config) (simpleStorageService, error) {
-	return s3.NewFromConfig(c), nil
+	var options func(options *s3.Options)
+	if c.BaseEndpoint != nil && *c.BaseEndpoint != "" {
+		options = func(options *s3.Options) {
+			options.UsePathStyle = true
+			options.BaseEndpoint = c.BaseEndpoint
+		}
+	}
+	return s3.NewFromConfig(c, options), nil
```

Comment:
I think `NewFromConfig` would end up calling the `nil` options in the case that the `BaseEndpoint` is not configured. We could have the options have a default of an empty function.

Reply:
@sorindumitru I do not understand what you mean. 
can you pls provide an example.

Reply:
The `options` variable defaults to the zero value of a function pointer, which is `nil`. `NewFromConfig` will then try to call that `nil` and get a segfault. See for example https://play.golang.com/p/SPZxT29tKjN

We can avoid this by initializing the variable to a function that doesn't do anything.

Reply:
@sorindumitru, yes you are right. 
@amartinezfayo  suggested to change it `var` [comment](https://github.com/spiffe/spire/pull/5757#discussion_r1945294092)
I'll revert it back to `options := func(options *s3.Options) {}`, is that OK?

Reply:
Sounds good to me.